File: Java/app/src/main/java/software/rsquared/template/di/AppComponent.java
package software.rsquared.template.di;

import android.app.Application;

import javax.inject.Singleton;

import dagger.BindsInstance;
import dagger.Component;
import dagger.android.AndroidInjectionModule;
import software.rsquared.template.ThisApp;

/**
 * @author rSquared.software
 */
@Singleton
@Component(modules = {
        AndroidInjectionModule.class,
        AppModule.class,
        ActivityBuilderModule.class
})
public interface AppComponent {

    @Component.Builder
    interface Builder {
        @BindsInstance
        Builder application(Application application);

        AppComponent build();
    }

    void inject(ThisApp app);

//	void inject(SomeService service);

}

File: app/src/main/java/software/rsquared/template/di/AppComponent.kt
package software.rsquared.template.di

import dagger.Component
import dagger.android.AndroidInjector
import dagger.android.support.AndroidSupportInjectionModule
import software.rsquared.template.ThisApp
import javax.inject.Singleton

/**
 * @author Rafal Orlik
 */
@Singleton
@Component(modules = [
	AndroidSupportInjectionModule::class,
	AppModule::class,
	ActivityBindingModule::class,
	ViewModelModule::class
])
interface AppComponent : AndroidInjector<ThisApp> {
	@Component.Builder
	abstract class Builder : AndroidInjector.Builder<ThisApp>()
}